packages/ethereum-contracts/contracts/superfluid/ConstantOutflowNFT.sol

```diff
@@ -0,0 +1,207 @@
+/// @dev This contract uses mint/burn interface for flow creation/deletion and holds the actual storage for both NFTs.
+contract ConstantOutflowNFT is CFAv1NFTBase {
+    /// @notice A mapping from token id to FlowData = { address sender, address receiver}
+    /// @dev The token id is uint256(keccak256(abi.encode(flowSender, flowReceiver)))
+    mapping(uint256 => FlowData) internal _flowDataByTokenId;
+
+    error COF_NFT_ONLY_CONSTANT_INFLOW();           // 0xa495a718
+    error COF_NFT_ONLY_CFA();                       // 0x054fae59
+    error COF_NFT_MINT_TO_AND_FLOW_RECEIVER_SAME(); // 0x0d1d1161
+    error COF_NFT_MINT_TO_ZERO_ADDRESS();           // 0x43d05e51
+    error COF_NFT_TOKEN_ALREADY_EXISTS();           // 0xe2480183
+    error COF_NFT_TRANSFER_IS_NOT_ALLOWED();        // 0x5b1855b1
+
+    function proxiableUUID() public pure override returns (bytes32) {
+        return
+            keccak256(
+                "org.superfluid-finance.contracts.ConstantOutflowNFT.implementation"
+            );
+    }
+
+    /// @notice An external function for querying flow data by `_tokenId``
+    /// @param _tokenId the token id
+    /// @return flowData the flow data associated with `_tokenId`
+    function flowDataByTokenId(
+        uint256 _tokenId
+    ) public view override returns (FlowData memory flowData) {
+        flowData = _flowDataByTokenId[_tokenId];
+    }
+
+    /// @note Neither mint nor burn will work here because we need to forward these calls.
+
+    /// @notice The mint function creates a flow from `_from` to `_to`.
+    /// @dev If `msg.sender` is not equal to `_from`, we `createFlowByOperator`.
+    /// Also important to note is that the agreement contract will handle the NFT creation.
+    /// @param _from desired flow sender
+    /// @param _to desired flow receiver
+    /// @param _flowRate desired flow rate
+    function mint(address _from, address _to, int96 _flowRate) external {
+        // regular create flow
+        if (msg.sender == _from) {
+            // superToken.createFlow(_to, _flowRate);
+        } else {
+            // superToken.createFlowFrom(_from, _to, _flowRate);
+        }
+    }
+
+    /// @notice The burn function deletes the flow between `sender` and `receiver` stored in `_tokenId`
+    /// @dev If `msg.sender` is not equal to `_from`, we `deleteFlowByOperator`.
+    /// Also important to note is that the agreement contract will handle the NFT deletion.
+    /// @param _tokenId desired token id to burn
+    function burn(uint256 _tokenId) external {
+        FlowData memory flowData = _flowDataByTokenId[_tokenId];
+        if (flowData.flowSender == msg.sender) {
+            // superToken.deleteFlow(flowData.sender, flowData.receiver);
+        } else {
+            // superToken.deleteFlowFrom(flowData.sender, flowData.receiver);
+        }
+    }
+
+    /// NOTE probably should be access controlled to only cfa
+    function onCreate(
```

Comment:
3a. These should look kinda familiar.
Looking at it now, it probably makes more sense to make all three take the exact same parameters: `address _flowSender, address _flowReceiver` for consistency.